### PR TITLE
nitcc: add prefixes

### PR DESCRIPTION
**Pull request by privat**
This add the `Prefixes` operator that returns the set of all prefixes of a regular language.
This also fixes some bugs.

### CODE REVIEW

File: contrib/nitcc/src/autom.nit [hunk 1]
```diff
@@ -390,32 +394,44 @@ class Automaton
...
 					continue
 				end
 			end
 		end
 
+		# Fixed point algorithm.
+		# * get 2 states s1 and s2 not yet distinguished
+		# * get a symbol w
+		# * if s1.trans(w) and s2.trans(w) are distinguished, then
+		#   distinguish s1 and s2
```
Comment (jcbrinfo) [inline contrib/nitcc/src/autom.nit]: A sentence should start with an upper case letter and end with a dot.

File: contrib/nitcc/src/autom.nit [hunk 2]
```diff
@@ -373,14 +373,18 @@ class Automaton
...
 		trim
 
+		# Graph of known distinct states
 		var distincts = new HashMap[State, Set[State]]
 		for s in states do
 			distincts[s] = new HashSet[State]
 		end
 
 		# split accept states
+		# An accept state is distinct with a non accept state
```
Comment (jcbrinfo) [inline contrib/nitcc/src/autom.nit]: Same thing here.

File: contrib/nitcc/src/autom.nit [hunk 3]
```diff
@@ -425,6 +441,8 @@ class Automaton
 			end
 		end
 
+		# We need, to unify not-distinguished states.
```
Comment (jcbrinfo) [inline contrib/nitcc/src/autom.nit]: > We need to unify not-distinguished states.

**Comment by jcbrinfo** [comment]
:+1: +1 (once the documentation mistakes are fixed)

**Comment by Morriar** [comment]
+1

**Pull request merged**